src/main/java/org/apache/mesos/scheduler/repair/TaskFailureListener.java

```diff
@@ -0,0 +1,17 @@
+package org.apache.mesos.scheduler.repair;
+
+import org.apache.mesos.Protos.TaskID;
+
+/**
+ * Exposes simple task-oriented API for components that need to interact with the framework as a whole.
+ */
+public interface TaskFailureListener {
+    /**
+     * Remove the given task from this framework. This ensures that the given task is not running, possibly by killing
+     * it, and it will remove the task from the framework's state tracking. The absence of the task will be fixed by the
+     * {@link RepairScheduler}, if that task is still supposed to be running.
```

Comment:
Do these comments still make sense?  Isn't a call back here an indication that the Task has is already not running?  Why any mention of removal from the Framework's state tracking?